app/src/store/user.ts

```diff
@@ -86,7 +107,7 @@ export const login = createAsyncThunk<
   {
     state: State;
     dispatch: Dispatch;
-    rejectValue: { reason: 'ID_TOKEN_UNDEFINED' | 'LOGIN_ERROR'; error: any };
+    rejectValue: { reason: 'ID_TOKEN_UNDEFINED' | 'LOGIN_ERROR'; error?: any };
```

Comment:
Good point, should be never. But as I was looking at all of our async action creators, I wanted them all to follow a similar pattern as such:

```ts
type RejectValue =
  | {
      reason: 'foo';
      error: SerializedError;
    }
  | {
      reason: 'bar';
    };
```

And I wanted to change all of our async creators to this pattern, but I think it should go into another ticket/PR since this one is so bloated already.

So making the error optional here for the `ID_TOKEN_UNDEFINED` reason is the smallest change I could think of that still is technically correct just for the sake of this PR. Ideally, the ticket I just mentioned is to clean up all the action creators so they follow the same pattern.